### ISSUE TITLE
mod_wsgi and importing _psycopg

### ISSUE THREAD

**Issue by edgarcosta** [psycopg/psycopg2#1144]
This may be related to #995 but I honestly don't understand what is going on in that thread.

When trying to import psycopg2 through mod_wsgi I get the same error message:
```
Traceback (most recent call last):
  File "/home/psetpartners/psetpartners-git-live/psetpartners.wsgi", line 1, in <module>
    import psycopg2
  File "/home/sage/sage-9.1/local/lib/python3.7/site-packages/psycopg2/__init__.py", line 51, in <module>
    from psycopg2._psycopg import (                     # noqa
SystemError: initialization of _psycopg raised unreported exception
```

I have checked shared object dependencies, via `ldd`, of the `.so` file no issues arise.
I have also noticed that the mod_wsgi documentation mentions some conflicts with psycopg2, but those seem to be outdated and only about reloading the app.

Any ideas of what ` initialization of _psycopg raised unreported exception` means?
I googled around, and the only relevant hits I found seem to be related to #995.

Any insight or help is appreciated.
Thank you in advance!

**Comment by dvarrazzo** [comment]
Hello. I don't have an idea on top of my mind about what it could be.

You may try to [compile psycopg2 with debug messages](https://www.psycopg.org/docs/install.html#creating-a-debug-build) and check if there are errors reported at import time that might suggest us where things get stuck.

If you would like us to try and investigate you should provide something to make the issue reproducible, such as a docker image.

Thank you very much!

**Comment by edgarcosta** [comment]
Thank you.
After some digging I figured out that it was mod_wsgi trying to use the default python environment despite buing compiled against a different python.

**Comment by othman-na** [comment]
How did you fix this ? @edgarcosta

**Comment by edgarcosta** [comment]
I returned to using gunicorn + proxy, as the python environment to use the app is delicate.

**Comment by juanrenos** [comment]
> 

Can you please detail the action to solve this? Thank you

**Comment by juanrenos** [comment]
> I returned to using gunicorn + proxy, as the python environment to use my app is delicate.


Can you please detail the action to solve this? Thank you

**Comment by edgarcosta** [comment]
Let me try again.
This issue is due to `mod_wsgi` using the system python, while my wsgi needs to use a different python environment (including binary), 
And thus, there are two solutions:
 - rebuild Apache (completely or just `mod_wsgi`) from the source to use the right linking
 - avoid `mod_wsgi` and use some other interface with the python app, for example, https://gunicorn.org/, which then you can access via [mod_proxy](https://httpd.apache.org/docs/2.4/mod/mod_proxy.html), e.g., https://serverfault.com/questions/627217/gunicorn-behind-apache-link-to-localhost

**Comment by dvarrazzo** [comment]
The sooner you get rid of mod_wsgi and everything that binds to apache, the better it is for you.
